### ISSUE TITLE
Improve migration robustness

### ISSUE THREAD

**Issue by nylen** [ClassicPress/ClassicPress#344]
We got a user report of a migration from WP -> ClassicPress failing to copy `composer.json` and then aborting.  This resulted in a broken install because some of the new functions that ClassicPress has defined in `wp-includes/version.php` were not available.

A few ideas to improve this situation:

- Don't overwrite `composer.json` if it already exists
- Identify files that are "non-critical" to a migration/upgrade, and allow the upgrade to succeed if they cannot be copied
- During migration to ClassicPress, copy `wp-includes/version.php` and other "critical" files first
- Add guards before using the new functions in other parts of the code (the error in this case was `Call to undefined function classicpress_is_dev_install() in .../wp-includes/script-loader.php:86`)

**Comment by Mte90** [comment]
For composer maybe is better a merge with the local one avalaible?

**Comment by nylen** [comment]
`Call to undefined function classicpress_is_dev_install() in .../wp-includes/script-loader.php:86`

I ran across this error on another site.  In this case it was due to a partial migration, but now that we've seen this a couple of times it seems like a good idea to prevent it.

**Comment by invisnet** [comment]
> An easier way to address this part of the issue could be to exclude `composer.json` from the ClassicPress builds used by the migration plugin. This would be done [here](https://github.com/ClassicPress/nightly-builder/blob/177f468d4efbc7a35158b453a43df056788e3ea5/build.sh#L63).

That would work, but unless I've missed something, doesn't it just push the problem to the first time they try to update CP? If we can't overwrite `composer.json` as part of the migration we won't be able to overwrite it as part of the update.

I think the right solution is to add another check to the migration plugin - it is, after all, a genuine conflict.

**Comment by invisnet** [comment]
PR is [here](https://github.com/ClassicPress/ClassicPress-Migration-Plugin/pull/53)

**Comment by nylen** [comment]
I merged the migration plugin PR to bail out if files that are **new to ClassicPress** already exist on the site.  Thanks @invisnet.

I think we can close this one now.  Re-reading what I wrote at the top...

> Add guards before using the new functions in other parts of the code

I'm thinking this isn't actually that good of an idea after all.  A broken site (in this case, a `wp-includes/version.php` file that is still from WordPress) should tell you exactly why it's broken in the error log, rather than trying to hide that fact without understanding other effects.

> During migration to ClassicPress, copy `wp-includes/version.php` and other "critical" files first

We can take another look at this when we know more about the specific ways this can fail, and any other effects a change like this would have.
